Sveta/task3 branch

Svitanok-auto opened the pull request.

Svitanok-auto committed:
Initial commit

Svitanok-auto committed:
Initial commit

Svitanok-auto committed:
Fixed some formatting

Svitanok-auto committed:
Changed local variable "sum" to be property "SumOfNumbersDevidableOn3AndNotDevidableOn5"

Svitanok-auto committed:
minor formatting

numash commented on Practice3/Program.cs:
```diff
@@ -1,15 +1,62 @@
 
 namespace Practice3
 {
     class Program
```
please, specify class access modifier

numash commented on Practice3/Program.cs:
```diff
@@ -1,15 +1,62 @@
     class Program
     {
-        static void Main(string[] args)
+        private static int LowerBorder { get; set; }
```
properties should be public

numash commented on Practice3/Validator.cs:
```diff
@@ -0,0 +1,49 @@
+                    Console.WriteLine("\nIncorrect Input: Empty value is not allowed");
+                }
+            }
+                return 0;
```
the alignment is broken a little bit

numash commented on Practice3/Validator.cs:
```diff
@@ -0,0 +1,49 @@
+
+namespace Practice3
+{
+    public static class Validator
```
the \n is missing somewhere

![image](https://user-images.githubusercontent.com/9382934/113509299-7b6a6700-955d-11eb-8eba-b9d8325db31a.png)

numash commented on Practice3/Program.cs:
```diff
@@ -1,15 +1,62 @@
+            }
+            else
+            {
+                Console.WriteLine("Less then 11 values range, exit");
```
I believe in this case the validation message is incorrect 

![image](https://user-images.githubusercontent.com/9382934/113509466-680bcb80-955e-11eb-8f3a-6bf70aab1aef.png)

numash replied:
not fixed for now

numash commented on Practice3/Program.cs:
```diff
@@ -1,15 +1,62 @@
+
+            if (IsRangeGoodToCreateArray(LowerBorder, UpperBorder))
+            {
+                int[] newArray = (int[])GetArrayOfNaturalNumbers(LowerBorder, UpperBorder);
```
you shouldn't change the type by (int[]) operation.
If you need this type, you should return it from the GetArrayOfNaturalNumbers method

numash commented on Practice3/Program.cs:
```diff
@@ -1,15 +1,62 @@
+            {
+                if ((array[i] % 3 == 0) && (array[i] % 5 != 0))
+                {
+                    SumOfNumbersDevidableOn3AndNotDevidableOn5 = SumOfNumbersDevidableOn3AndNotDevidableOn5 + array[i];
```
please, don't use the global class property for this operation. the local variable would be better here.

numash commented on Practice3/Program.cs:
```diff
@@ -1,15 +1,62 @@
+
+        private static void GetSumOfNumbersDevidableOn3AndNotDevidableOn5(int [] array)
+        {
+            for (int i = 0; i < UpperBorder - LowerBorder; i++)
```
It makes more sense to use array length instead of the "UpperBorder - LowerBorder"

Svitanok-auto committed:
Changes per review comments were done

numash commented on Practice3/Validator.cs:
```diff
@@ -0,0 +1,49 @@
+                        }
+                        else
+                        {
+                            Border = Convert.ToInt32(inputString);
```
please, don't use the class property as local variable

Svitanok-auto committed:
Updates are:
1 Validation for negative input is added;
2 Removed Border property and added local variable border;

Svitanok-auto committed:
removed brackets
